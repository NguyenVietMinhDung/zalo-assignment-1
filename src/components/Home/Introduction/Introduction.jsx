import React from 'react';
import { Container, Row, Col } from 'react-grid-system';
import SearchBox from './SearchBox';
import Metadata from './Metadata';
import ShopSelector from './ShopSelector';
import './Introduction.scss';
import banner from '../../../assets/images/BANNER.png';

const Introduction = () => (
  <Container>
    <Row className="introduction">
      <Col className="introduction__banner" xs={12} style={{ padding: 0 }}>
        <img src={banner} alt="" />
      </Col>

      <Col className="introduction__search-box" xs={12} md={4} lg={3} offset={{ md: 8, lg: 9 }}>
        <SearchBox />
      </Col>

      <Col className="introduction__metadata" xs={12}>
        <Metadata />
      </Col>

      <Col xs={12}>
        <ShopSelector />
      </Col>
    </Row>
  </Container>
);

export default Introduction;
