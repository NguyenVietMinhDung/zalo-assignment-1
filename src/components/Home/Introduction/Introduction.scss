.introduction {
	margin-bottom: 1px;

	color: black;
	background-color: white;

	&__banner {
		padding-right: -15px;
		padding-left: -15px;

		img {
			width: 100%;
			height: 100%;
		}
	}

	&__search-box {
		margin-top: 10px;
	}

	&__metadata {
		&__title {
			display: flex;

			i {
				margin-top: 1.3em;
				margin-right: 5px;

				color: green;

				font-weight: bold;
			}

			h3 {
				margin-bottom: 0;
			}
		}

		&__address,
		&__time {
			display: flex;

			i {
				margin-top: 1em;
				margin-right: 5px;

				opacity: 0.2;

				font-weight: bold;
			}

			p {
				margin-bottom: 0;
			}
		}
	}
}
