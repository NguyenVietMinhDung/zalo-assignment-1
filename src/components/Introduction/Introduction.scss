.introduction {
  color: black;
  margin-bottom: 1px;

  &__banner {
    padding-left: -15px;
    padding-right: -15px;

    img {
      width: 100%;
      height: 100%;
    }
  }

  &__metadata {
    background-color: white;

    &__title {
      display: flex;

      i {
        margin-right: 5px;
        margin-top: 1.3em;
        color: green;
        font-weight: bold;
      }

      h3 {
        margin-bottom: 0;
      }
    }

    &__address, &__time {
      display: flex;

      i {
        margin-right: 5px;
        margin-top: 1em;
        opacity: 0.2;
        font-weight: bold;
      }

      p {
        margin-bottom: 0;
      }
    }
  }
}
