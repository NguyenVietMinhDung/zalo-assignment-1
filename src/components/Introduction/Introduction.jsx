import React from 'react';
import { Container, Row, Col } from 'react-grid-system';
import './Introduction.scss';
import banner from '../../assets/images/BANNER.png';

const Introduction = () => (
  <Container>
    <Row className="introduction">
      <Col className="introduction__banner" xs={12} style={{ padding: 0 }}>
        <img src={banner} alt="" />
      </Col>

      <Col className="introduction__metadata" xs={12}>
        <div className="introduction__metadata__title" style={{ color: 'black' }}>
          <i className="lni lni-16 lni-checkmark-circle" />
          <h3>Highlands Coffee - 123 Nguyễn Đình Chiểu</h3>
        </div>
        <div className="introduction__metadata__address">
          <i className="lni lni-16 lni-map-marker" />
          <p>Lô G15 Tầng trêt, Trung tâm thương mại Aeon Tân Phú, 30 Bờ Bao Tân Thắng, Phường Sơn Kỳ, Quận Tân Phú, Tp.Hồ Chí Minh</p>
        </div>
        <div className="introduction__metadata__time">
          <i className="lni lni-16 lni-timer" />
          <p>08:00 - 22:00</p>
        </div>
      </Col>
    </Row>
  </Container>
);

export default Introduction;
