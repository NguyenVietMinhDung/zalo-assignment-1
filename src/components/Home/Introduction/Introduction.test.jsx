import React from 'react';
import { shallow } from 'enzyme';
import Introduction from './Introduction';

describe('<Introduction />', () => {
  test('renders correctly', () => {
    const component = shallow(<Introduction />);
    expect(component).toMatchSnapshot();
  });
});
