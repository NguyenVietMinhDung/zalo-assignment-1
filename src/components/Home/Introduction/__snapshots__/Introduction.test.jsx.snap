// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Introduction /> renders correctly 1`] = `
<Container
  component="div"
  fluid={false}
  lg={false}
  md={false}
  sm={false}
  style={Object {}}
  xl={false}
  xs={false}
  xxl={false}
>
  <Row
    align="normal"
    className="introduction"
    component="div"
    debug={false}
    gutterWidth={null}
    justify="start"
    nogutter={false}
    nowrap={false}
    style={Object {}}
  >
    <Col
      className="introduction__banner"
      component="div"
      debug={false}
      lg={null}
      md={null}
      offset={Object {}}
      pull={Object {}}
      push={Object {}}
      sm={null}
      style={
        Object {
          "padding": 0,
        }
      }
      width={null}
      xl={null}
      xs={12}
      xxl={null}
    >
      <img
        alt=""
        src="test-file-stub"
      />
    </Col>
    <Col
      className="introduction__search-box"
      component="div"
      debug={false}
      lg={3}
      md={4}
      offset={
        Object {
          "lg": 9,
          "md": 8,
        }
      }
      pull={Object {}}
      push={Object {}}
      sm={null}
      style={Object {}}
      width={null}
      xl={null}
      xs={12}
      xxl={null}
    >
      <SearchBox />
    </Col>
    <Col
      className="introduction__metadata"
      component="div"
      debug={false}
      lg={null}
      md={null}
      offset={Object {}}
      pull={Object {}}
      push={Object {}}
      sm={null}
      style={Object {}}
      width={null}
      xl={null}
      xs={12}
      xxl={null}
    >
      <Metadata />
    </Col>
    <Col
      component="div"
      debug={false}
      lg={null}
      md={null}
      offset={Object {}}
      pull={Object {}}
      push={Object {}}
      sm={null}
      style={Object {}}
      width={null}
      xl={null}
      xs={12}
      xxl={null}
    >
      <ShopSelector />
    </Col>
  </Row>
</Container>
`;
